src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/FilterResult.cs

```diff
@@ -17,5 +18,51 @@ public FilterResult(CompletionItem completionItem, string filterText, bool match
             MatchedFilterText = matchedFilterText;
             FilterText = filterText;
         }
+
+        public static int Compare(FilterResult result1, FilterResult result2)
+        {
+            var item1 = result1.CompletionItem;
+            var item2 = result2.CompletionItem;
+
+            var prefixLength1 = item1.FilterText.GetCaseInsensitivePrefixLength(result1.FilterText);
+            var prefixLength2 = item2.FilterText.GetCaseInsensitivePrefixLength(result2.FilterText);
+
+            // Prefer the item that matches a longer prefix of the filter text.
+            if (prefixLength1 != prefixLength2)
+            {
+                return prefixLength1 > prefixLength2 ? 1 : -1;
```

Comment:
Not sure about performance considerations, but:
```suggestion
                return prefixLength1.CompareTo(prefixLength2);
```